### DISCUSSION TITLE
Blender Exporter - Why is "Nav-mesh" extension not ending up in the GLTF file?

### DISCUSSION THREAD

**Discussion by emclaren** [mozilla/hubs#2694] [General Questions & Discussion] [answered]
When I use the Blender exporter, the "nav-mesh" extension doesn't end up in the GLTF file. Instead I just get an empty MOZ_hubs_components extension added to that node. Am I missing a trick?

Note: this question was transferred to Github Discussions from our Discord server.

**Comment by netpro2k** [comment, marked as answer]
Hmm, that component was mostly in there for internal testing, it probably does not work correctly yet. We should probably ship witha more restrictive component configuration that only includes "supported" components.

Currently the solution for custom navmeshes in particular is to bring in the mesh you want to use for navigation into spoke, set it as the only `walkable` item, and hide it. You can then regenerate the floorplan and will only be based on your custom mesh. Its not quite 100%, since its still voxelizing but it will get you pretty close.

**Reply by rawnsley** [reply]
Thanks for the suggestion. The voxelizing isn't great on undulating terrain where the difference between the nav-mesh and visual polygons is jarring, particularly in VR. 

For context in hoping for an All-Blender workflow ultimately. For now I'll revert to hand editing the GLTF file.
